### ISSUE TITLE
Configurable HTTP server timeout parameters

### ISSUE THREAD

**Issue by leonardoce** [cloudnative-pg/cloudnative-pg#630]
In https://github.com/cloudnative-pg/cloudnative-pg/commit/197b3f23daab610d5796dcea8e3a8f551ef32e95  we introduced timeouts on the http.Server as a security measure against the SlowLoris type of attack.

While this may be totally correct, we are using the HTTP server to upload the new instance manager executable from the operator Pod to the PostgreSQL Pod and the timeouts we have now, which is hardcoded to 30 seconds, may not be enough depending on the network configuration of that Kubernetes cluster.

This parameters should be configurable at least at operator level.

**Comment by sxd** [comment]
I'm not sure this needs to go in the Cluster spec section, since it's more like an operator/instance manager configuration and not part of one cluster. What do you think @leonardoce @hh24k ?

**Comment by leonardoce** [comment]
Yes, at first glance I think this don't belong to the Cluster resource but it's more an operator configuration, but I'm sure there's some valid reason to have it in the Cluster resource too.

Why we don't create a new issue to put them in the Cluster API and for now we rely on just the operator configuration?

**Comment by sxd** [comment]
@leonardoce yes, that's what I think because probably we will need to create a new section in the clusters for this kind of configuration

**Comment by hh24k** [comment]
> I'm not sure this needs to go in the Cluster spec section, since it's more like an operator/instance manager configuration and not part of one cluster. What do you think @leonardoce @hh24k ?

@sxd I was thinking of cases when some clusters may fall in regions with a worse network so that it needs bigger timeouts to be customizable in Cluster spec. ... but as @leonardoce suggested, we can consider this scenario more carefully in another issue.

**Comment by gbartolini** [comment]
I suggest we limit to the operator configuration level now, without touching the API for the cluster (which could be part of another ticket).
